src/EventFeed.AspNetCore/EventFeedMiddleware.cs

```diff
@@ -3,79 +3,100 @@
 using Microsoft.Extensions.DependencyInjection;
 using EventFeed.Producer.Abstractions;
 using System.Text.Json;
+using System.Text.Json.Serialization;
+using EventFeed.AspNetCore.Serialization;
 
 namespace EventFeed.AspNetCore
 {
     public class MetaLinks
     {
-        public MetaLinks(string _meta, string _head, string _tail)
+        public MetaLinks(string meta, string head, string tail)
         {
-            this._meta = _meta;
-            this._head = _head;
-            this._tail = _tail;
+            Meta = meta;
+            Head = head;
+            Tail = tail;
         }
 
-        public string _meta { get; private set; }
-        public string _head { get; private set; }
-        public string _tail { get; private set; }
+        [JsonPropertyName("_meta")]
+        public string Meta { get; private set; }
+        [JsonPropertyName("_head")]
+        public string Head { get; private set; }
+        [JsonPropertyName("_tail")]
+        public string Tail { get; private set; }
     }
 
     public class Links
     {
-        public Links(string _meta, string _head, string _previous, string _self, string _next, string _tail)
+        public Links(string meta, string head, string previous, string self, string next, string tail)
         {
-            this._meta = _meta;
-            this._head = _head;
-            this._previous = _previous;
-            this._self = _self;
-            this._next = _next;
-            this._tail = _tail;
+            Meta = meta;
+            Head = head;
+            Previous = previous;
+            Self = self;
+            Next = next;
+            Tail = tail;
         }
 
-        public string _meta { get; private set; }
-        public string _head { get; private set; }
-        public string _previous { get; private set; }
-        public string _self { get; private set; }
-        public string _next { get; private set; }
-        public string _tail { get; private set; }
+        [JsonPropertyName("_meta")]
+        public string Meta { get; private set; }
+        [JsonPropertyName("_head")]
+        public string Head { get; private set; }
+        [JsonPropertyName("_previous")]
+        public string Previous { get; private set; }
+        [JsonPropertyName("_self")]
+        public string Self { get; private set; }
+        [JsonPropertyName("_next")]
+        public string Next { get; private set; }
+        [JsonPropertyName("_tail")]
+        public string Tail { get; private set; }
```

Comment:
I am just going to leave some comments to point out the cost of idiomatic.

- 6 Lines become 12
- you need to look at not the property name but a string in an attribute 15 or so characters above it and to the right when reading. So instead of scanning properties. You are scanning prop, attr-> to the right, prop, attr-> to the right
- renaming, refactoring is a 2 step process for a DTO, the thing that IS the contract